tests/testthat/test-core_functions.R

```diff
@@ -231,6 +290,66 @@ test_that("`get_transition_value()` works with a predictor that varies over time
   expect_equal(result, 11:20)
 })
 
+test_that("`get_transition_value()` works with a predictor that varies over time
+  and a duration-based transition", {
+    # Arrange
+    t <- transition(
+      from = "a",
+      to = "b",
+      fun = function(x, y) x * y,
+      transition_type = "duration",
+      predictors = list(x = predictor_spec("temp", first_day_only = FALSE)),
+      parameters = parameters(y = c(day = 1, night = 2))
+    )
+
+    predictors <- predictors(data.frame(
+      pred = "temp",
+      # in this example, the subcategories could correspond to day and night temp
+      pred_subcategory = c("day", "night"),
+      j_day = sort(rep(1:2, 2)),
+      value = c(10, 2, 8, 1)
+    ))
+
+
+    # Act
+    result <- get_transition_value(
+      time = 1,
+      transition = t,
+      population = empty_population_matrix(c("a", "b"), 10L),
+      developing_population = empty_population_matrix(c("a", "b"), 10L),
+      max_duration = 365L,
+      predictors = predictors
+    )
+
+    # Assert
+
+    # Result is a vector with an element for each (day, subcategory) pair. So in
+    # this case, we get a vector of length 4 even though we only have a 2 day
+    # period. This is partially a result of the transition function - e.g. if
+    # we use `function(x, y) sum(x * y)` we get a scalar return value.
+    # TODO we might want to be stricter about the lengths of vectors that
+    # transition functions can return. This decision should consider how results
+    # from get_transition_value() are used - in get_transition_matrix and
+    # update_delay_array.
+    expect_identical(result, c(day = 10, night = 4, day = 8, night = 2))
```

Comment:
See this note and the next FIXME test. In a separate PR, we may want to dig more into these possible cases

Reply:
Sorry this wasn't clear - frankly I'm still trying to make sense of this too. A concrete example is the test on lines [337-353](https://github.com/dallenmidd/IxPopDyMod/blob/73bc10825f01325890a9146bf2c498926a2f7bd8/tests/testthat/test-core_functions.R#L337-L353). 

More broadly, I think that there are two cases we want to allow:
1. Probability-based transitions should return a scalar value
2. Duration-based transitions should return a vector of length matching the number of days the transition is supposed to last. (Currently, if a duration-based transition returns a scalar, we `rep()` it so we can do a `cumsum()`, but conceptually we always want to get a longer vector from a duration-based transition).

The issue is that there's no code that enforces this, because a transition function can be basically any function. It just has to have an argument for each predictor and parameter specified in that transition. But a user could supply a function that returns anything (a vector of length 2 in my linked example). So, I think we'd probably want a check in `get_transition_value()` or where that's called, to see that the length of the returned value makes sense.

Hope this is clearer... it's definitely (too!) complicated and I'm still trying to wrap my head around it. I think it'd be worth addressing as it could lead to pretty opaque/confusing bugs - if you agree I can make this into an issue and take a stab. On the upside, I don't think it would involve any changes to existing configs, just the `core_functions.R` code.